### PR TITLE
update of Scripts `à la carte`

### PR DESCRIPTION
**Pull request by nick-youngblut**
Provided explicit examples of using input values & paths in a la carte scripts.

Signed-off-by: Nick Youngblut <nyoungb2@gmail.com>

### CODE REVIEW

File: docs/process.rst [hunk 1]
```diff
@@ -134,20 +134,29 @@ To use a language other than Bash, simply start your process script with the cor
...
+        output:
+        path "table.txt"
+    
         """
         #!/usr/bin/python
-
-        x = 'Hello'
-        y = 'world!'
-        print "%s - %s" % (x,y)
+
```
Comment (mribeirodantas) [inline docs/process.rst:156]:
```suggestion
        // Reads the content of a file, and writes it to `table.txt`
```

**Review by mribeirodantas — CHANGES_REQUESTED** [review]
For beginners, this can be slightly difficult to read so I would suggest to add a comment, such as this one.

**Review by mribeirodantas — APPROVED** [review]

File: docs/process.rst [hunk 2]
```diff
@@ -134,20 +134,29 @@ To use a language other than Bash, simply start your process script with the cor
...
+        output:
+        path "table.txt"
+    
         """
         #!/usr/bin/python
-
-        x = 'Hello'
-        y = 'world!'
-        print "%s - %s" % (x,y)
+        // Reads the content of a file, and writes it to `table.txt`
```
Comment (mribeirodantas) [inline docs/process.rst:156]:
```suggestion
        # Reads the content of a file, and writes it to `table.txt`
```

It's Python now, not Groovy/Nextflow so we should use the appropriate comment symbol.

**Review by mribeirodantas — CHANGES_REQUESTED** [review]

File: docs/process.rst [hunk 3]
```diff
@@ -134,20 +134,29 @@ To use a language other than Bash, simply start your process script with the cor
 `shebang <http://en.wikipedia.org/wiki/Shebang_(Unix)>`_. For example::
 
     process perlStuff {
+        input:
+        val(x)
```
Comment (mribeirodantas) [inline docs/process.rst:138]:
```suggestion
        val(x)
        output:
        stdout
```

File: docs/process.rst [hunk 4]
```diff
@@ -134,20 +134,29 @@ To use a language other than Bash, simply start your process script with the cor
...
         #!/usr/bin/perl
 
-        print 'Hi there!' . '\n';
+        print 'value: ${x}';
         """
     }
 
     process pythonStuff {
+        input:
+        path(x)
```
Comment (mribeirodantas) [inline docs/process.rst:149]:
```suggestion
        path(x)
        output:
        stdout
```

**Review by mribeirodantas — CHANGES_REQUESTED** [review]

**Review by mribeirodantas — APPROVED** [review]

**Comment by pditommaso** [comment]
Hi, sorry for the late reply. The goal of the example is to show how to invoke other interpreters via the shebang declaration. 

For this reason, I think it's better to add input and keep the example self-contained. Changed  it to add the workflow invocation 6d6b3543

**Comment by nick-youngblut** [comment]
@pditommaso if the goal of this example was to:

>show how to invoke other interpreters via the shebang declaration

...then maybe including a separate example in the docs of how to actually write python/perl code using shell variable interpretation (as in this PR) would be helpful?

**Comment by pditommaso** [comment]
Likely. The key thing it's not going to be a Shell variable, but it should use the syntax of the target scripting language ie. python/perl/R etc. I think this is what confuses people

**Pull request closed without merging**